src/diffusion.py

```diff
@@ -205,15 +205,45 @@ def compute_mmd(x, y):
     x_kernel = compute_kernel(x, x)
     y_kernel = compute_kernel(y, y)
     xy_kernel = compute_kernel(x, y)
-    mmd = x_kernel.mean() + y_kernel.mean() - 2*xy_kernel.mean()
+    mmd = x_kernel + y_kernel - 2*xy_kernel
     return mmd
 
-def get_loss(x,y, params, dcg, FD, model):
+
+class weighted_loss():
+    def __init__(self, y, weight = "n"):
+        self.y = y
```

Comment:
The loggings need to be removed.
Secondly, I understand "n" and "sqrt" mode, but what if we want to do unweighted training? In that case any mode other than "n" and "sqrt" will work which does not seem good to me. Instead make an option to have weight == "False" which will trigger training with unweighted mode. If mode is neither "n", "sqrt" or "False" raise a KeyError.

Reply:
My rationale for that is I want the program to compile with any value for weight. I have mentioned in the Readme that the suggested value for weight to obtain the unweighted loss is `weight='None'`. If you really think that the program should crash for any other value of `weight`, I can do that.

Reply:
It will not crash for any random value of weight like`weight = 'random_string'`. Instead for a random value of weight it will just train without weighted loss which doesn't make sense to me. Instead I think only `weight == 'None'` should trigger program to do unweighted loss.

Reply:
Fixed in https://github.com/vishnu-m77/DiffAPTOSC/pull/47/commits/6467a2becd43daadb2f831b4d32e44daa3b96a85